### PR TITLE
Random operator selection scheme

### LINKED ISSUE #97: Introduce RandomSelect scheme
Introduce a selection scheme that selects destroy and repair operators with uniform probability. To be done after introducing the new selection interface.

### PR DESCRIPTION
**Pull request by N-Wouda**
Closes #97. This PR introduces RandomSelect.

### CODE REVIEW

**Comment by N-Wouda** [comment]
@leonlan this is ready for review!

File: alns/select/RandomSelect.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        p = np.sum(self.op_coupling, axis=1) / np.sum(self.op_coupling)
+        d_idx = rnd.choice(np.arange(self.num_destroy), p=p)
+
+        r_indices = np.flatnonzero(self.op_coupling[d_idx, :])
+        r_idx = rnd.choice(r_indices)
+
+        return d_idx, r_idx
+
+    def update(self, candidate, d_idx, r_idx, s_idx):
+        pass  # pragma: no cover
```
Comment (N-Wouda) [inline alns/select/RandomSelect.py:27]: This is a no-op. There's no point in having that show up in the coverage report.

File: alns/select/RandomSelect.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    """
+
+    def __call__(self, rnd, best, curr):
+        """
+        Selects a destroy operator with uniform probability. Then selects a
+        repair operator with uniform probability, respecting the operator
+        coupling matrix.
+        """
+        p = np.sum(self.op_coupling, axis=1) / np.sum(self.op_coupling)
+        d_idx = rnd.choice(np.arange(self.num_destroy), p=p)
```
Comment (leonlan) [inline alns/select/RandomSelect.py:19]: The docstring describes something different than the implementation :-). Do we want to sample each combination (d_op, r_op) uniformly or do we want to sample d_op and r_op separately?

File: alns/select/RandomSelect.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import numpy as np
+
+from alns.select.OperatorSelectionScheme import OperatorSelectionScheme
+
+
+class RandomSelect(OperatorSelectionScheme):
```
Comment (leonlan) [inline alns/select/RandomSelect.py:6]: RandomSelect is not yet added to the docs.

File: alns/select/RandomSelect.py (hunk 2 above)
Comment (leonlan) [reply alns/select/RandomSelect.py:19]: I think the current implementation makes sense. When there's no op coupling then the operators are all sampled uniformly and independently. If there is op coupling, then we should look at the operator pairs instead and sample those uniformly.

So changing the docstring works for me.

Comment (leonlan) [reply alns/select/RandomSelect.py:19]: Suggestion: maybe we can make this even easier? Find all non-zero (i, j) in the op_coupling matrix. Then apply rnd.choice on those index tuples (i, j)?

**Comment by N-Wouda** [comment]
Great, I think that covers it. Really nice simplified selection!

**Review by leonlan — APPROVED** [review]

**Pull request merged**